packages/ui-extensions-sdk/src/utils/security.ts

```diff
@@ -1,19 +1,28 @@
 import { LoadedResourceMetaDataBatch, NetworkRequestMetadata } from '../types';
 
-// TODO: Feel free to pass in whatever data structure you need
-// to track what has been previously loaded. Here I'm assuming it will be a set of
-// string ids
 export type LoadedResourceIds = Set<string>;
 
 export const collectResourceUsage = (
     previouslyLoadedResources: LoadedResourceIds
 ): [LoadedResourceMetaDataBatch, LoadedResourceIds] => {
-    const newLoadedResourceIds = new Set(previouslyLoadedResources);
+    // Use resource timing api to to retrieve application loaded resources
 
-    // TODO: Do batch collection of resource data
-    // TODO: Update list of preivously loaded resource ids
+    const snapshot = performance.getEntriesByType('resource').map(e => {
+        const a = e as PerformanceResourceTiming;
+        return {
+            url: a.name,
+            initiatorType: a.initiatorType,
```

Comment:
If the initiator type is `FETCH` let's remove the path from the URL for privacy reasons.  Not sure where you can inject that logic?  Maybe in the interface?  Should we consider any other `initiatorType`?

Reply:
`xmlhttprequest` and `fetch` seems to be the only two request-related initiator types.
Also RUM is doing similar filtering in : 
https://github.com/DataDog/browser-sdk/blob/cb1918e8a94a9fdabef9fcc3396a2bd0ddf8ad62/packages/rum-core/src/domain/rumEventsCollection/resource/resourceUtils.ts#L74-L76